x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -258,4 +260,7 @@ export const parseModelStateFromStats = (
 export const parseModelStateReasonFromStats = (trainedModelStats?: Partial<MlTrainedModelStats>) =>
   trainedModelStats?.deployment_stats?.reason;
 
-export const getMlInferencePrefixedFieldName = (fieldName: string) => `ml.inference.${fieldName}`;
+export const getMlInferencePrefixedFieldName = (fieldName: string) =>
+  `ml.inference.${stripMlInferencePrefix(fieldName)}`; // Strip first, then prepend, to prevent against double-prepending
```

Comment:
suggestion: What if you check whether it starts with the target string before modifying? Then you can do:
```
export const getMlInferencePrefixedFieldName = (fieldName: string) => fieldName.startsWith('ml.inference') ? fieldName : 
  `ml.inference.${fieldName}`;
```